src/app/public/modules/modal/modal-instance.ts

```diff
@@ -11,48 +11,106 @@ import {
   }
 
+  /**
+   * An event that the modal instance emits when it closes.
+   * It emits a `SkyCloseModalArgs` object with a `data` property that includes
+   * data passed from users on close or save and a `reason` property that indicates
+   * whether the modal was saved or closed without saving.
+   * The `reason` property accepts any string value.
+   * Common examples include `cancel`, `close`, and `save`.
+   */
   public get closed(): Observable<SkyModalCloseArgs> {
     return this._closed;
   }
 
+  /**
+   * An event that the modal instance emits when users click
+   * the <i class="fa fa-question-circle" aria-hidden="true"></i> button.
+   * If a `helpKey` parameter was specified, the `helpOpened` event broadcasts the `helpKey`.
+   */
   public get helpOpened(): Observable<string> {
     return this._helpOpened;
   }
 
+  /**
+   * A direct reference to the provided component's class.
+   */
+  public componentInstance: any;
+
   private _beforeClose = new Subject<SkyModalBeforeCloseHandler>();
 
   private _closed = new Subject<SkyModalCloseArgs>();
 
   private _helpOpened = new Subject<string>();
 
-  public close(result?: any, reason?: string, ignoreBeforeClose?: boolean) {
+  /**
+   * Closes the modal instance.
+   * @param result Specifies an object that will be emitted to
+   * subscribers of the `closed` event of the modal instance.
+   * The `SkyModalInstance` provider can be injected into a component's constructor
+   * so that this `close` function can be called from a button in the `sky-modal-footer`.
+   * @param reason Specifies the reason for the modal closing, with the default reason of `close`.
+   * @param ignoreBeforeClose Ignores the modal instance's `beforeClose` event.
+   */
+  public close(result?: any, reason?: string, ignoreBeforeClose?: boolean): void {
     if (reason === undefined) {
       reason = 'close';
     }
 
     this.closeModal(reason, result, ignoreBeforeClose);
   }
 
-  public cancel(result?: any) {
+  /**
+   * Closes the modal instance with `reason=cancel`.
+   * @param result Specifies an object that will be emitted to subscribers
+   * of the closed event of the modal instance. The SkyModalInstance provider
+   * can be injected into a component's constructor so that this cancel function can be called
+   * from a button in the sky-modal-footer.
+   */
+  public cancel(result?: any): void {
     this.closeModal('cancel', result);
   }
 
-  public save(result?: any) {
+  /**
+   * Closes the modal instance with `reason=save`.
+   * @param result Specifies an object that will be emitted to subscribers of the
+   * `closed` event of the modal instance. The `SkyModalInstance` provider can be injected
+   * into a component's constructor so that this save function can be called
+   * from a button in `the sky-modal-footer`.
+   */
+  public save(result?: any): void {
     this.closeModal('save', result);
   }
 
-  public openHelp(helpKey?: string) {
+  /**
+   * Triggers the helpOpened event.
```

Comment:
I think `helpOpened` should be in tick marks to make it a code snippet. Also, should we explain the event here? i.e. "Triggers the `helpOpened` event that broadcasts a `helpKey` parameter to open when users click the ? button." Not sure if we need that here, but it seems like it would be useful.

Reply:
I'm going to update the `?` to `<i class="fa fa-question-circle" aria-hidden="true"></i>`